### PR TITLE
New Rules for AutoIt3.exe behaving in a way matching the DG infection chain

### PR DESCRIPTION
**Pull request by mbabinski**
### Summary of the Pull Request

Adding two new rules within the emerging threats/malware section covering TTPs observed in recent research and reporting on DarkGate malware. DarkGate seems to use AutoIt3.exe in it's infection chain. This is true for the VBS and MSI infection methods. These rules detect AutoIt3 process and file creation that matches what I have seen in reports and my own research testing DarkGate VBS/MSI samples.

### Changelog

new: DarkGate - Autoit3.EXE File Creation By Uncommon Process
new: DarkGate - Autoit3.EXE Execution Parameters
update: DarkGate - User Created Via Net.EXE

### Example Log Event

File creation events:
image
Process creation events:
image

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

File: rules-emerging-threats/2023/Malware/DarkGate/file_event_win_malware_darkgate_autoit3.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+title: Autoit3.exe Executable File Creation Matching DarkGate Behavior
```
Comment (nasbench) [inline rules-emerging-threats/2023/Malware/DarkGate/file_event_win_malware_darkgate_autoit3.yml:1]:
```suggestion
title: DarkGate - Autoit3.EXE File Creation By Uncommon Process
```

File: rules-emerging-threats/2023/Malware/DarkGate/file_event_win_malware_darkgate_autoit3.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            - '\curl.exe'
+            - '\KeyScramblerLogon.exe'
+            - '\Autoit3.exe'
+            - '\wmprph.exe'
+            - '\ExtExport.exe'
+        TargetFilename|endswith: '\Autoit3.exe'
+    condition: selection
+falsepositives:
+    - Unknown
+level: medium
```
Comment (nasbench) [inline rules-emerging-threats/2023/Malware/DarkGate/file_event_win_malware_darkgate_autoit3.yml:36]:
```suggestion
level: medium

```

File: rules-emerging-threats/2023/Malware/DarkGate/file_event_win_malware_darkgate_autoit3.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    category: file_event
+    product: windows
+detection:
+    selection:
+        Image|endswith:
+            - '\curl.exe'
+            - '\KeyScramblerLogon.exe'
+            - '\Autoit3.exe'
+            - '\wmprph.exe'
+            - '\ExtExport.exe'
```
Comment (nasbench) [inline rules-emerging-threats/2023/Malware/DarkGate/file_event_win_malware_darkgate_autoit3.yml:31]:
```suggestion
            - '\Autoit3.exe'
            - '\curl.exe'
            - '\ExtExport.exe'
            - '\KeyScramblerLogon.exe'
            - '\wmprph.exe'
```

File: rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+title: DarkGate Autoit3.exe Execution Parameters
```
Comment (nasbench) [inline rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml:1]:
```suggestion
title: DarkGate - Autoit3.EXE Execution Parameters
```

File: rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+detection:
+    selection_proc:
+        - Image|endswith: '\Autoit3.exe'
+        - OriginalFileName: 'AutoIt3.exe'
+    selection_parent:
+        ParentImage|endswith:
+            - '\cmd.exe'
+            - '\KeyScramblerLogon.exe'
+            - '\msiexec.exe'
+    filter_normal_autoit3:
```
Comment (nasbench) [inline rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml:30]:
```suggestion
    filter_main_legit_autoit_location:
```

File: rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        Image: 'C:\Program Files (x86)\AutoIt3\AutoIt3.exe'
```
Comment (nasbench) [inline rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml:31]:
```suggestion
        Image|endswith: ':\Program Files (x86)\AutoIt3\AutoIt3.exe'
```

File: rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    condition: all of selection_* and not filter_normal_autoit3
+falsepositives:
+    - Unlikely
+level: high
```
Comment (nasbench) [inline rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml:35]:
```suggestion
level: high

```

File: rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        - Image|endswith: '\Autoit3.exe'
+        - OriginalFileName: 'AutoIt3.exe'
+    selection_parent:
+        ParentImage|endswith:
+            - '\cmd.exe'
+            - '\KeyScramblerLogon.exe'
+            - '\msiexec.exe'
+    filter_normal_autoit3:
+        Image: 'C:\Program Files (x86)\AutoIt3\AutoIt3.exe'
+    condition: all of selection_* and not filter_normal_autoit3
```
Comment (nasbench) [inline rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_appdata.yml:32]:
```suggestion
    condition: all of selection_* and not 1 of filter_main_*
```

**Comment by nasbench** [comment]
Thanks for the contribution @babinski

Unfortunately I'm AFK for now but can you please add a README.md similar to the other emerging threat rules? That would be helpful and appreciated. Here is an example https://github.com/SigmaHQ/sigma/blob/7364ce00b1444802caaebefdccb6d8f6a92f1112/rules-emerging-threats/2023/Malware/COLDSTEEL/README.md

It's fine if you don't. I'll add it when I'm back.

Thanks 👍

**Comment by nasbench** [comment]
Thanks for the contribution @babinski

Unfortunately I'm AFK for now but can you please add a README.md similar to the other emerging threat rules? That would be helpful and appreciated. Here is an example https://github.com/SigmaHQ/sigma/blob/7364ce00b1444802caaebefdccb6d8f6a92f1112/rules-emerging-threats/2023/Malware/COLDSTEEL/README.md

It's fine if you don't. I'll add it when I'm back.

Thanks 👍

**Review by nasbench — APPROVED** [review]

File: rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_susp_parent_and_location.yml [hunk 9]
```diff
@@ -0,0 +1,35 @@
+title: DarkGate - Autoit3.EXE Execution Parameters
+id: f8e9aa1c-14f2-4dbd-aa59-b98968ed650d
+status: experimental
+description: |
+    Detects execution of the legitimate Autoit3 utility from a suspect parent process. AutoIt3.exe is used within
```
Comment (phantinuss) [inline rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_susp_parent_and_location.yml:5]:
```suggestion
    Detects execution of the legitimate Autoit3 utility from a suspicious parent process. AutoIt3.exe is used within 
```

File: rules-emerging-threats/2023/Malware/DarkGate/file_event_win_malware_darkgate_autoit3_binary_creation.yml [hunk 10]
```diff
@@ -0,0 +1,36 @@
+title: DarkGate - Autoit3.EXE File Creation By Uncommon Process
+id: 1a433e1d-03d2-47a6-8063-ece992cf4e73
+status: experimental
+description: |
+    Detects the usage of curl.exe, KeyScramblerLogon, or other non-standard/suspicious processes used to create Autoit3.exe. 
+    This activity has been associated with DarkGate malware, which uses Autoit3.exe to execute shellcode that performs 
+    process injection and connects to the DarkGate command-and-control server. Curl, KeyScramblerLogon, and these other 
+    processes consitute non-standard and suspicious ways to retrieve the Autoit3 executable.
+references:
+    - Internal Research
```
Comment (phantinuss) [inline rules-emerging-threats/2023/Malware/DarkGate/file_event_win_malware_darkgate_autoit3_binary_creation.yml:10]:
```suggestion
```

File: rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_susp_parent_and_location.yml [hunk 11]
```diff
@@ -0,0 +1,35 @@
+title: DarkGate - Autoit3.EXE Execution Parameters
+id: f8e9aa1c-14f2-4dbd-aa59-b98968ed650d
+status: experimental
+description: |
+    Detects execution of the legitimate Autoit3 utility from a suspect parent process. AutoIt3.exe is used within 
+    the DarkGate infection chain to execute shellcode that performs process injection and connects to the DarkGate 
+    command-and-control server.
+references:
+    - Internal Research
```
Comment (phantinuss) [inline rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_susp_parent_and_location.yml:9]:
```suggestion
```

File: rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_susp_parent_and_location.yml [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    selection_proc:
+        - Image|endswith: '\Autoit3.exe'
+        - OriginalFileName: 'AutoIt3.exe'
+    selection_parent:
+        ParentImage|endswith:
+            - '\cmd.exe'
+            - '\KeyScramblerLogon.exe'
+            - '\msiexec.exe'
+    filter_main_legit_autoit_location:
+        Image|endswith: ':\Program Files (x86)\AutoIt3\AutoIt3.exe'
```
Comment (phantinuss) [inline rules-emerging-threats/2023/Malware/DarkGate/proc_creation_win_malware_darkgate_autoit3_from_susp_parent_and_location.yml:31]:
```suggestion
        Image|endswith:
           - ':\Program Files (x86)\AutoIt3\AutoIt3.exe'
           - ':\Program Files\AutoIt3\AutoIt3.exe'
```

**Review by phantinuss — APPROVED** [review]

**Pull request merged**